src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/XamlRequestExecutionQueue.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Xaml.Telemetry;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml.LanguageServer
+{
+    internal class XamlRequestExecutionQueue : RequestExecutionQueue<RequestContext>, ILspService
+    {
+        private readonly XamlProjectService _projectService;
+        private readonly IXamlLanguageServerFeedbackService? _feedbackService;
+
+        public XamlRequestExecutionQueue(
+            XamlProjectService projectService,
+            IXamlLanguageServerFeedbackService? feedbackService,
+            AbstractLanguageServer<RequestContext> languageServer,
+            ILspLogger logger,
+            IHandlerProvider handlerProvider) : base(languageServer, logger, handlerProvider)
+        {
+            _projectService = projectService;
+            _feedbackService = feedbackService;
+        }
+
+        public override async Task<TResponseType> ExecuteAsync<TRequestType, TResponseType>(
```

Comment:
moved unchanged for the most part.

Reply:
for my edification, how does xaml differ generally here?

Reply:
There are a couple extra xaml services that they call for each LSP request, one that does some kind of document tracking and one that collects separate telemetry.  Not sure exactly what the details are, would have to ask Marco.

Reply:
Right, the XamlProjectService keeps track of the XAML shadow project and manages the XAML Documents in there. And the feedback service aggregates telemetry in VS for us.